From 207f8eaeb1ede98576d279236d8afb55c041147c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <bostik@odin.bostik.iki.fi>
Date: Mon, 12 Dec 2011 09:29:26 +0200
Subject: [PATCH 05/13] Use load operator for atomic int

API for QAtomicInt has changed since 4.x - there is no longer a
specialised gt/lt inequality operator, but there is a specific .load()
to look up the value of the given atomic int.
---
 src/threed/scene/qglrenderstate.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/threed/scene/qglrenderstate.cpp b/src/threed/scene/qglrenderstate.cpp
index 2e3da94..a4dca94 100644
--- a/src/threed/scene/qglrenderstate.cpp
+++ b/src/threed/scene/qglrenderstate.cpp
@@ -281,7 +281,7 @@ void QGLRenderState::detach()
     }
     else
     {
-        if (d->ref > 1)  // being shared, must detach
+        if (d->ref.load() > 1)  // being shared, must detach
         {
             QGLRenderStatePrivate *temp = d->clone();
             d->ref.deref();
-- 
1.7.7.3

